packages/x-components/src/x-modules/facets/components/filters/all-filter.vue

```diff
@@ -36,56 +36,68 @@
    *
    * @public
    */
-  @Component({
-    components: { BaseEventButton },
-    mixins: [xComponentMixin(facetsXModule)]
-  })
-  export default class AllFilter extends Vue {
-    /** The facet data. */
-    @Prop({ required: true })
-    public facet!: Facet;
-
-    /** The getter of the selectedFiltersByFacet. */
-    @Getter('facets', 'selectedFiltersByFacet')
-    public selectedFiltersByFacet!: FiltersByFacet;
-
-    /**
-     * The event that will be emitted when the all filter button is clicked.
-     *
-     * @returns The event to emit on click.
-     * @internal
-     */
-    protected get clickEvent(): Partial<XEventsTypes> {
-      return {
-        UserClickedAllFilter: [this.facet.id]
-      };
-    }
-
-    /**
-     * Computed to retrieve the selected state of this component.
-     *
-     * @returns True if is selected false otherwise.
-     */
-    protected get isSelected(): boolean {
-      return isArrayEmpty(this.selectedFiltersByFacet?.[this.facet.id]);
-    }
+  export default defineComponent({
+    name: 'AllFilter',
+    components: {
+      BaseEventButton
+    },
+    xModule: facetsXModule.name,
+    props: {
+      /** The facet data. */
+      facet: {
+        type: Object as PropType<Facet>,
+        required: true
+      }
+    },
+
+    setup(props) {
+      /** The getter of the selectedFiltersByFacet. */
+      const { selectedFiltersByFacet }: Dictionary<ComputedRef<FiltersByFacet>> = useGetter(
+        'facets',
+        ['selectedFiltersByFacet']
+      );
+
+      /**
+       * The event that will be emitted when the all filter button is clicked.
+       *
+       * @returns The event to emit on click.
+       * @internal
+       */
+      const clickEvent = computed((): Partial<XEventsTypes> => {
+        return { UserClickedAllFilter: [props.facet.id] };
+      });
+
+      /**
+       * Computed to retrieve the selected state of this component.
+       *
+       * @returns True if is selected false otherwise.
+       */
+      const isSelected = computed((): boolean => {
```

Comment:
`boolean` return type is already inferred and you can use the implicit return if you want
```TS
const isSelected = computed(() =>
  isArrayEmpty(selectedFiltersByFacet.value?.[props.facet.id])
);
```